Source/WebCore/platform/graphics/transforms/TransformState.h

```diff
@@ -113,6 +117,7 @@ class TransformState {
     std::optional<FloatQuad> mappedSecondaryQuad(bool* wasClamped = nullptr) const;
 
     TransformationMatrix* accumulatedTransform() const { return m_accumulatedTransform.get(); }
+    std::unique_ptr<TransformationMatrix> releaseTrackedTransform() { return WTFMove(m_trackedTransform); }
```

Comment:
why not use release()?

Reply:
Hmm, to avoid confusion: what does "TransformState::release()" tell you? :-) releaseTransform() is not an improvement either, I think we do need a suffix to differentiate 'tracked transform' vs. 'accumulated transform'.

However this is not a strong opinion. I just think "release" is too generic.

Reply:
Sorry, I meant release() on unique_ptr.

Reply:
Ah. I didn't even think about release() on unique_ptr, since that gives me a naked pointer. We really want to move the unique_ptr and its content together elsewhere -- therefore WTFMove(m_trackedTransform) seems appropriate.

Reply:
We further discussed this. m_trackedTransform is probably 'undefined' after moving out of TransformState. So calling translate() / etc. afterwards while e.g. TrackScreenCTMMatrix is enabled, would lead to undefined behavior / corruption / crash. (Or did I miss something in c++20 maybe that protects us here? I memorized using-object-after-move is a red flag).

However, the call-site that makes use of the new method calls "releaseTrackingTransform()" as last operation, before the TransformState would be destructed anyhow -- so we're just efficiently moving out the unique_ptr<TransformationMatrix> out of TransformState before it would get destructed.

Still a bit fragile in general IMHO -- however other users in WebCore use the same pattern, e.g. takeStyle() from StyleResolver. Just leaving this as comment, that this topic was visited, and we came to the conclusion that the code is fine as-is, but this needs to be checked case-by-case.